### DISCUSSION TITLE
Player Housing / Decorative Item Placement

### DISCUSSION THREAD

**Discussion by LuxPoseidon** [EveronLife/EveronLife#60] [Ideas]
Add the ability for apartments to be rented and houses to be owned. May need to attach an obj to the main door that would give the prompt. 

On top of this system the ability for a player to buy Décor from a shop and place it in their home would be cool, they could even buy storage to increase the size of a virtual storage made for their home. Apartments would have less storage then a house by default and obviously has more space.

**Comment by thebonbon** [comment]
As explained in https://github.com/EveronLife/EveronLife/discussions/61 I also added a very basic building system. It was designed to also support systems like this. 
I will look into it more and make a little poc.

**Comment by DrLuk** [comment]
In Arma 3 it was problematic in terms of performance, but I remember some servers having a system where you could place your house somewhere (as a ruin) and then bring certain materials like bricks, cement, glass etc. to "build" it yourself. I think that would be awesome, but we might need to look into the performance side of things.

**Reply by LuxPoseidon** [reply]
Performance shouldn't be an issue, the map already has every building fleshed out with furniture, paintings, etc.

Arma 3 was a CPU locked game with issues at its core engine. Reforger should play just fine with it however as always we will code and make things to be as performant as possible regardless.

**Comment by ThibaudetTevai** [comment]
I made a mod which :

- Buy a slot for the house
- Get the materials to build the house
- Add the house in the database to be respawn when the server restart.

[https://www.youtube.com/watch?v=l3QZIS0DCws&ab_channel=TunuiRichmond](url)

Anyone on this idea ?

**Reply by LuxPoseidon** [reply]
Would like to see what you can make. Not so sure on the having to wait for a restart for the building.

I know it’s a lot of work but building states would be cool, first have the buyer of a plot of land pick a prefab for the building then give it materials like some concrete and then it makes a slab, then you provide some boards and it make a frame of the walls etc.
